### PR TITLE
Added support and documentation for Trusted Types

### LINKED ISSUE #157: Adding Support for Trusted Types
### Background
Trusted Types is a relatively new web application security feature that protects against DOM XSS. Trusted Types enforcement is currently supported in Chrome 83 and Android Webview. Even where Trusted Types are not enforced by the browser they can still be used to reduce the DOM XSS attack surface.

**_DOM XSS_**
Cross site scripting (XSS) is one of the most prevalent web security vulnerabilities. DOM XSS occurs on the client side when Javascript calls dangerous functions with user supplied input. Examples of dangerous sink functions include innerHTML, outerHTML, eval, and setTimer. Nonce-based CSP already helps prevent server-side XSS. Trusted types help prevent DOM-based or client-side XSS caused by these sink functions. 

**_How It Works_**
Trusted Types are enabled using CSP directives. When enabled this way, the browser will require user supplied data to be processed before being sent to a sink function. Sink functions can only be called with special trusted types objects that tell the browser the data has been sanitized and should be trusted. There are three types of trusted objects; TrustedHTML, TrustedScript, and TrustedScriptURL which are used and created depending on what object type a sink expects.

Trusted objects are created by calling policies. Policies are functions that sanitize input, enforce security rules, and return a trusted type object. Trusted objects can also be created by some libraries like DOMPurify.

If enabled with CSP’s report only mode, browsers will report Trusted Types violations but not block them.

For more information on Trusted Types visit [here](https://web.dev/trusted-types/) and [here](https://w3c.github.io/webappsec-trusted-types/dist/spec/#integration-with-content-security-policy).

### Proposed Changes to Django-CSP
**_Add require-trusted-types-for Directive_**
To enable Trusted Types with CSP two new directives are required. The require-trusted-types-for directive should be configurable using Django-CSP. This directive enables Trusted Types by specifying for which groups of injection sinks trusted types should be expected. Only one value currently exists for this directive, ‘script’, which requires Trusted Types for DOM XSS injection sinks.

**_Add trusted-types Directive_**
The trusted-types directive should be configurable using Django-CSP. This directive can be used to control the creation of Trusted Types policies. Only policies with names present in this directive can be created. It also allows for the creation/declaration of a default policy. When this directive is not present, policies cannot be created; so if trusted types are expected, injection sinks can not be used at all. 

**_Additional Documentation_**
These two new CSP directives should be added to the Configuring Django-CSP documentation. A new page should be added to the documentation providing more information on Trusted Types and steps for how to build a compliant web application.

### PR DESCRIPTION
**Pull request by meggles711**
Fixes #157

### CODE REVIEW

**Comment by pmac** [comment]
I'm not an expert on the trusted types directive, but the code looks good to me. Anyone else want to give a review for the docs and direction? @g-k ?

File: docs/trusted_types.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+===================================
+
+``DOM Cross-site Scripting``
+============================
+Cross-site scripting (XSS) is one of the most prevalent vulnerabilities
+on the web. Nonce-based CSP is used to prevent server-side XSS. Trusted
+Types are used to prevent client-side or DOM-XSS_. Trusted Types rely on
+the browser to enforce the policy that is provided to it. Currently, Trusted
+Types are supported on Chrome 83 and Android Webview. Many browsers are in the
+process of adding support. Check back for updated compatibility_.
```
Comment (g-k) [inline docs/trusted_types.rst:12]: nit: would be good to mention the polyfill too https://github.com/w3c/webappsec-trusted-types#polyfill

**Comment by pmac** [comment]
There are two trusted types PRs. This one and #158. This one has docs but that one was first. I appreciate all of the contributions but I'm not sure how best to proceed.

**Comment by g-k** [comment]
CI is failing on #157 with a line length pep8 nit. I'm in favor of landing this one since CI is passing and it includes docs.

**Comment by 9mido** [comment]
@g-k and @pmac  Please be sure to upload these newest PRs to PyPi.

**Comment by pmac** [comment]
I'll be looking into doing that today.

**Comment by pmac** [comment]
Version 3.7 is now on PyPI

**Pull request merged**